src/status_im2/common/standard_auth/view.cljs

```diff
@@ -0,0 +1,174 @@
+(ns status-im2.common.standard-auth.view
+  (:require [react-native.core :as rn]
+            [quo2.components.numbered-keyboard.numbered-keyboard.view :as numbered-keyboard]
+            [quo2.core :as quo]
+            [status-im2.common.standard-auth.style :as style]
+            [reagent.core :as reagent]
+            [quo2.foundations.colors :as colors]
+            [react-native.fast-image :as fast-image]
+            [status-im2.common.resources :as resources]
+            [utils.i18n :as i18n]
+            [quo2.components.keycard.view :as keycard]
+            [utils.re-frame :as rf]))
+
+(defn enter-keycard-pin-sheet
+  []
+  (let [entered-numbers (reagent/atom [])
+        max-digits 6
+        max-attempt-reached (reagent/atom false)
+
+        handle-number-entry
+        (fn [number]
+          (when (>= (count @entered-numbers) max-digits)
+            (reset! entered-numbers []))
+          (when (= (count @entered-numbers) (dec max-digits))
+            (reset! max-attempt-reached true))
+          (swap! entered-numbers conj number))
+        get-dot-style
+        (fn [idx]
```

Comment:
imo, this function would be better to have in the styles file, it makes it a bit confusing to see what styles are set. just make styles/digit a function instead 👍

Reply:
@J-Son89 Created a function in styles files for this purpose and used it in view file.